internal/output/utils.go

```diff
@@ -75,13 +87,126 @@ func MakeTable(rows []Row) ([]string, []Row) {
+				width[i] = len(sv)
+			}
+			if !alignSet {
+				align[i] = alignmentForType(v.Type)
+			}
 		}
+		alignSet = true
 		rows[ri] = row
 	}
 	// remove this. prefix from header cells
 	thisPfx := fmt.Sprintf("%s.", NameValue)
+	columns := make(table.Row, len(header))
 	for i, h := range header {
-		header[i] = strings.TrimPrefix(h, thisPfx)
+		columns[i] = table.Column{
+			Header: strings.TrimPrefix(h, thisPfx),
+			Align:  align[i] * width[i],
+		}
+	}
+	return columns, rows
+}
+
+func makeTableHeaderFromRow(row Row, maxWidth int) table.Row {
+	dw := defaultWidthForFlexibleColumns(row, maxWidth)
+	// the max width for a column is 1000, just an arbitrary decision --YT
+	if dw > 1000 {
+		dw = 1000
 	}
-	return header, rows
+	ww := 0
+	for _, c := range row {
+		ww += widthForType(c.Type, len(c.Name), dw)
+	}
+	hd := make(table.Row, len(row))
+	for i, c := range row {
+		hd[i] = table.Column{
+			Header: c.Name,
+			Align:  alignmentForType(c.Type) * widthForType(c.Type, len(c.Name), dw),
+		}
+	}
+	return hd
+}
+
+func alignmentForType(t int32) int {
+	switch t {
+	case serialization.TypeByte, serialization.TypeUInt16, serialization.TypeInt16,
+		serialization.TypeInt32, serialization.TypeInt64,
+		serialization.TypeFloat32, serialization.TypeFloat64,
+		serialization.TypeJavaBigInteger, serialization.TypeJavaDecimal:
+		return -1
+	default:
+		return 1
+	}
+}
+
+func widthForType(t int32, headerWidth, defaultWidth int) int {
+	switch t {
+	case serialization.TypeByte:
+		return maxInt(headerWidth, len(fmt.Sprint(math.MaxInt8)))
+	case serialization.TypeUInt16, serialization.TypeInt16:
+		return maxInt(headerWidth, len(fmt.Sprint(math.MaxInt16)))
+	case serialization.TypeInt32:
+		return maxInt(headerWidth, len(fmt.Sprint(math.MaxInt32)))
+	case serialization.TypeInt64:
+		return maxInt(headerWidth, len(fmt.Sprint(math.MaxInt64)))
+	case serialization.TypeFloat32:
+		return maxInt(headerWidth, len(fmt.Sprint(math.MaxFloat32)))
+	case serialization.TypeFloat64:
+		return maxInt(headerWidth, len(fmt.Sprint(math.MaxFloat64)))
+	case serialization.TypeBool:
+		return maxInt(headerWidth, len(fmt.Sprint(false)))
+	case serialization.TypeUUID:
+		return maxInt(headerWidth, len(fmt.Sprint(types.UUID{})))
+	default:
+		return maxInt(headerWidth, defaultWidth)
+	}
+}
+
+func defaultWidthForFlexibleColumns(header Row, maxWidth int) int {
+	if maxWidth < 1 {
+		return 1
+	}
+	const padding = 3
+	fixedSize := 0
+	flexibleCount := 0
+	for _, h := range header {
+		switch h.Type {
+		case serialization.TypeByte, serialization.TypeUInt16, serialization.TypeInt16,
+			serialization.TypeInt32, serialization.TypeInt64,
+			serialization.TypeFloat32, serialization.TypeFloat64,
+			serialization.TypeBool, serialization.TypeUUID:
+			fixedSize += widthForType(h.Type, len(h.Name), len(h.Name)) + padding
+		default:
+			flexibleCount++
+		}
+	}
+	if flexibleCount == 0 {
+		return 1
+	}
+	w := (maxWidth-fixedSize)/flexibleCount - padding
+	if w < 0 {
```

Comment:
When we return 1, is this means that for this column we only show it 1 width? I did not understand what will show if it is 1.
Maybe we can set a lower limit, if the value is lower than this we may not show this column.

Reply:
> When we return 1, is this means that for this column we only show it 1 width?
That's right. 

> Maybe we can set a lower limit, if the value is lower than this we may not show this column.
I have a different idea about hiding columns, something like: "display only x columns, and display `...` for the rest" if there's no way to fit the column to the width (console or set by user). But that's not implemented yet.